src/commands/executeFunction/eventGrid/EventGridSourceStep.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, type IAzureQuickPickItem } from "@microsoft/vscode-azext-utils";
+import { localize } from "../../../localize";
+import { type EventGridExecuteFunctionContext } from "./EventGridExecuteFunctionContext";
+import { supportedEventGridSourceLabels, supportedEventGridSources, type EventGridSource } from "./eventGridSources";
+
+export class EventGridSourceStep extends AzureWizardPromptStep<EventGridExecuteFunctionContext> {
+    public hideStepCount: boolean = false;
+
+    public async prompt(context: EventGridExecuteFunctionContext): Promise<void> {
+        // Prompt for event source
+        const eventGridSourcePicks: IAzureQuickPickItem<EventGridSource | undefined>[] = supportedEventGridSources.map((source: EventGridSource) => {
+            return {
+                label: supportedEventGridSourceLabels.get(source) || source,
+                data: source,
+            };
+        });
+        const eventSource: EventGridSource =
+            (
+                await context.ui.showQuickPick(eventGridSourcePicks, {
+                    placeHolder: localize('selectEventSource', 'Select the event source'),
+                    stepName: 'eventGridSource',
+                })
+            ).data ?? 'Microsoft.Storage';
```

Comment:
Why fallback to Microsoft.Storage? Do we expect the data to return `undefined` for any reason?

Reply:
I'm not actually sure what scenario causes this to be undefined, but I saw that it's a possible scenario in the typing. This is a leftover from a rougher prototype tbh. What should be the appropriate scenario here?

Reply:
We should probably just throw an error here honestly since it isn't likely to happen.

I think ignoring the undefined and autofilling it with storage would be a really weird behavior for the user to decipher what the heck just happened lol